core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -119,17 +119,19 @@ public ManifestGroup select(String... columns) {
               manifest.addedFilesCount() + manifest.existingFilesCount() > 0);
     }
 
-    Iterable<Iterable<ManifestEntry>> readers = Iterables.transform(
+    Iterable<CloseableIterable<ManifestEntry>> readers = Iterables.transform(
         matchingManifests,
         manifest -> {
           ManifestReader reader = ManifestReader.read(ops.io().newInputFile(manifest.path()));
           FilteredManifest filtered = reader.filterRows(dataFilter).select(columns);
           toClose.add(reader);
```

Comment:
This seems redundant, can we also get rid of the `toClose()` field?

Reply:
Good catch. Removing it.